src/components/grid/GridContainer.tsx

```diff
@@ -45,26 +46,27 @@ const GridContainer: FC<GridContainerProps> = (props: GridContainerProps) => {
   };
 
   const onLayoutChange = (newLayouts: Layout[]) => {
-    setLayouts(newLayouts);
+    setWidgetConfigs(getWidgetConfByLayout(newLayouts, widgetConfigs));
   };
+
   const isOnSettingPage = useLocation().pathname === RouterPaths.SETTINGS;
   /**
    * fire when a widget is moved and placed in a different position
    * @param newLayouts
    */
   const onDragStop = async (newLayouts: Layout[]) => {
-    setLayouts(newLayouts);
```

Comment:
IO during a dragdrop will slow down the UI. In my view it is better to keep the latest widget config related to the layout in the localstate/memory and flush it to firebase periodically using a background process - probably using a Saga. It is easy to keep the actual layout and the memory based widget config in sync. When app is closed (Do we get an event when this happen?) we can flush the latest config if dirty flag is set. The worse that can happen in this case is we fail to flush to firebase the latest config so next time the dashboard is loaded user will see an older config. For this type of an app that is perfectly fine. 
Please consider re-implementing this using Redux/Saga.